utils/manifest-utils/src/keystore/x509_keystore.rs

```diff
@@ -29,17 +31,24 @@ pub(super) trait X509AddParams {
     fn whole_chain(&self) -> bool;
 }
 
+#[derive(Eq, PartialEq)]
 pub struct X509CertData {
     pub id: String,
-    pub not_after: String,
+    pub not_after: DateTime<Utc>,
     pub subject: BTreeMap<String, String>,
     pub permissions: String,
 }
 
 impl X509CertData {
     pub fn create(cert: &X509Ref, permissions: &Vec<CertPermissions>) -> anyhow::Result<Self> {
         let id = cert_to_id(cert)?;
-        let not_after = cert.not_after().to_string();
+        // Openssl lib allows to access time only through ASN1_TIME_print.
+        // Diff starting from epoch is a workaround to get `not_after` value.
+        let time_diff = Asn1Time::from_unix(0)?.diff(cert.not_after())?;
+        let not_after = NaiveDateTime::from_timestamp_millis(0).unwrap()
+            + Duration::days(time_diff.days as i64)
+            + Duration::seconds(time_diff.secs as i64);
+        let not_after = DateTime::<Utc>::from_utc(not_after, Utc);
```

Comment:
There is a test for date format which executes this code https://github.com/golemfactory/yagna/pull/2515/files#diff-ec6a39728ba2e4c152bdfa5d2656c52af494157fefaa9392fd40b4205a24f658R250
(I know it expires in 2025, but I did not knew how to generate new set of Golem certificates).
`Asn1Time::from_unix(0)` will not fail, and `.diff(cert.not_after())?` is an openssl's `ASN1_TIME_diff` called on value generated by [X509_getm_notAfter](https://www.openssl.org/docs/manmaster/man3/X509_getm_notAfter.html). 
I am not sure if it will be possible to generate X509 certificate which `openssl` will be able to parse and it will make this code fail.

Reply:
Added some base unit to cover before/after epoch date conversion.